cmd/set/set_test.go

```diff
@@ -0,0 +1,89 @@
+package set
```

Comment:
There may be opinions on testing in this way, notably:

- Performing tests of `set` subcommands via the top level command
- The necessity to pass the `Config` struct in and having a mock instance of this.

Happy to discuss a different approach, however, the reasoning was that the arg sanitation is all being done in `cmd/set/puppet` rather than in `config` - to test this was working, I needed to see how this was being called.

Other potential approaches:

- Push arg sanitation / processing to the `config` package?
- Just recall values from Viper?

I still think there's value in this approach as it'll make testing `(get | set) backend` easier.

Reply:
needs to be `set_test` package